### DISCUSSION TITLE
How can I add Git Bash?

### DISCUSSION THREAD

**Discussion by xianzheTM** [kingToolbox/WindTerm#707] [Q&A] [answered]
1. How can I add Git Bash?
2. Can you set the default shell session? I don't want to use CMD.
version: 2.4.1

**Comment by lcmgithub** [comment, marked as answer]
new session -> shell -> session -> shell -> choose gitbash

**Reply by xianzheTM** [reply]
![image](https://user-images.githubusercontent.com/25103583/167527975-962c8c13-4372-4374-b180-1876bae7eb96.png)
No git bash

**Reply by lcmgithub** [reply]
Whether the git command is added to the system `path` variable

**Reply by xianzheTM** [reply]
Yes,I have add it.

**Reply by lcmgithub** [reply]
`.wind\profiles\default.v10\terminal` directory `user.sessions` file added
```
{
"process.arguments" : "--login -i",
"process.workingDirectory" : "$(HomeDir)",
"session.group" : "Shell sessions",
"session.icon" : "session::gitbash",
"session.label" : "gitbash",
"session.protocol" : "Shell",
"session.system" : "linux",
"session.target" : "C:/Program Files/Git/bin/bash.exe",
"session.uuid" : "2a587793-5d72-4944-b0a4-140e3a927264",
"window.caretStyle" : "blinking line"
}
```

**Reply by xianzheTM** [reply]
It's work,thx.

**Reply by wieniawski-lai** [reply]
> `.wind\profiles\default.v10\terminal` directory `user.sessions` file added
> 
> ```
> {
> "process.arguments" : "--login -i",
> "process.workingDirectory" : "$(HomeDir)",
> "session.group" : "Shell sessions",
> "session.icon" : "session::gitbash",
> "session.label" : "gitbash",
> "session.protocol" : "Shell",
> "session.system" : "linux",
> "session.target" : "C:/Program Files/Git/bin/bash.exe",
> "session.uuid" : "2a587793-5d72-4944-b0a4-140e3a927264",
> "window.caretStyle" : "blinking line"
> }
> ```

Hi,

I have done as you said, but I still failed to find git-bash in shell:
![image](https://user-images.githubusercontent.com/70927596/223003726-191537f6-1708-4405-962e-b17ec285bbbd.png)

And the user.config is 
```json
{
    "application.language" : "zh-CN",
    "terminal.reimportShellSessions" : false,
    "process.arguments" : "--login -i",
    "process.workingDirectory" : "$(HomeDir)",
    "session.group" : "Shell sessions",
    "session.icon" : "session::gitbash",
    "session.label" : "gitbash",
    "session.protocol" : "Shell",
    "session.system" : "linux",
    "session.target" : "D:\un-setup\Git\git-bash.exe",
    "session.uuid" : "2a587793-5d72-4944-b0a4-140e3a927264",
    "window.caretStyle" : "blinking line"
}
```
And also the git-bash.exe is added too path
![image](https://user-images.githubusercontent.com/70927596/223004114-7be6db0e-83c8-4379-8e94-362c568d61dd.png)

and the location of git-bash.exe is 
![image](https://user-images.githubusercontent.com/70927596/223004092-828f3561-78a0-47ff-a440-b59375eea56a.png)

Any idea why does not work?

thx

**Reply by xianzheTM** [reply]
It is not "git-bash.exe", but /bin/git.exe in the "git installation location".

**Comment by wieniawski-lai** [comment]
thx

first of all, I guess it should be "bash.exe", instead of "git.exe"

And I changed my settings to
```json
{
    "process.arguments" : "--login -i",
    "process.workingDirectory" : "$(HomeDir)",
    "session.group" : "Shell sessions",
    "session.icon" : "session::gitbash",
    "session.label" : "gitbash",
    "session.protocol" : "Shell",
    "session.system" : "linux",
    "session.target" : "D:/un-setup/Git/bin/bash.exe",
    "session.uuid" : "2a587793-5d72-4944-b0a4-140e3a927264",
    "window.caretStyle" : "blinking line"
}
```

and enviroment variable is
![image](https://user-images.githubusercontent.com/70927596/223293953-8acbd3b9-c31a-40c3-82a8-55764b9ffd12.png)

but I still failed to find git-bash in windterm
![image](https://user-images.githubusercontent.com/70927596/223294071-0f034b9a-543c-4d6a-a6c5-4f5f90bbef68.png)

**Comment by wieniawski-lai** [comment]
en,

If I choose custom and set like this
![image](https://user-images.githubusercontent.com/70927596/223295178-aae2e065-e1f5-4e0d-9412-9aa5b3910d30.png)

I can log in the git-bash successfully.

However, how can I save it and make it to the default shell?

thx.

**Reply by xianzheTM** [reply]
Do you mean you can't save the session and can only connect via "custom" each time?
Set default shell: Preferences -> Settings ->Application -> Startup -> A specific session or set of sessions

**Reply by wieniawski-lai** [reply]
> Do you mean you can't save the session and can only connect via "custom" each time? Set default shell: Preferences -> Settings ->Application -> Startup -> A specific session or set of sessions

perfect

that solves my problem

thx
